backend/src/main/resources/graphql/schema.graphqls

```diff
@@ -1,8 +1,14 @@
 type Query {
     userById(userID: ID!): User
+    getAllUsers: [User]!
+
     promptById(promptID: ID!): Prompt
 }
 
+type Mutation {
+    addUser(name: String!): User
```

Comment:
does `: User` mean it always returns a User, or does GQL have some kind of syntax for optional returning, as in there's a difference between `: User` and `: User!`?

Reply:
If so, this would be `: User!` because it always returns a user, while something like `userById` would be `: User` since it might not find that user?

Reply:
its actually the exact opposite of what you wrote, if it has a bang then it has to be there and without it it could be null. not sure what the case would be for it being null? perhaps a persistence error? unsure tbh

Reply:
oh, then, this should be `User!`? if it fails, i don't think graphql cares (otherwise everything can possibly fail / nothing is guaranteed). I'd put a bang here in any case